application/src/main/java/org/togetherjava/tjbot/commands/filesharing/FileSharingMessageListener.java

```diff
@@ -0,0 +1,183 @@
+ * {@link FileSharingMessageListener#extensionFilter}
+ */
+public class FileSharingMessageListener extends MessageReceiverAdapter {
+
+    private static final ObjectMapper JSON = new ObjectMapper();
+
+    private final String gistApiKey;
+    private static final String SHARE_API = "https://api.github.com/gists";
+    private static final HttpClient CLIENT = HttpClient.newHttpClient();
+
+    private final Set<String> extensionFilter = Set.of("txt", "java", "gradle", "xml", "kt", "json",
+            "fxml", "css", "c", "h", "cpp", "py", "yml");
+
+    private final Predicate<String> isStagingChannelName;
+    private final Predicate<String> isOverviewChannelName;
+
+
+    public FileSharingMessageListener(@NotNull Config config) {
+        super(Pattern.compile(".*"));
+
+        gistApiKey = config.getGistApiKey();
+        isStagingChannelName = Pattern.compile(config.getHelpSystem().getStagingChannelPattern())
+            .asMatchPredicate();
+        isOverviewChannelName = Pattern.compile(config.getHelpSystem().getOverviewChannelPattern())
+            .asMatchPredicate();
+    }
+
+    @Override
+    public void onMessageReceived(@NotNull MessageReceivedEvent event) {
+        User author = event.getAuthor();
+        if (author.isBot() || event.isWebhookMessage()) {
+            return;
+        }
+
+        if (!isHelpThread(event)) {
+            return;
+        }
+
+        List<Message.Attachment> attachments =
+                event.getMessage().getAttachments().stream().filter(attachment -> {
+                    String extension = attachment.getFileExtension();
+                    if (extension == null) {
+                        return false;
+                    }
+                    return extensionFilter.contains(extension);
+                }).toList();
+
+        processAttachments(event, attachments);
+    }
+
+    private void processAttachments(@NotNull MessageReceivedEvent event,
+            @NotNull List<Message.Attachment> attachments) {
+        Map<String, GistFile> filesAsJson = new HashMap<>();
+
+        for (Message.Attachment attachment : attachments) {
+            attachment.retrieveInputStream()
+                .thenApply(this::readAttachment)
+                .thenAccept(content -> filesAsJson.put(createFileName(attachment),
+                        new GistFile(content)))
+                .join();
+        }
+
+        GistFiles files = new GistFiles(filesAsJson);
+        GistRequest request = new GistRequest(event.getAuthor().getName(), false, files);
+        String url = uploadToGist(request);
+        sendResponse(event, url);
+    }
+
+    private @NotNull String readAttachment(@NotNull InputStream stream) {
+        try {
+            return new String(stream.readAllBytes(), StandardCharsets.UTF_8);
+        } catch (IOException e) {
+            throw new UncheckedIOException(e);
+        }
+    }
+
+    private @NotNull String createFileName(@NotNull Message.Attachment attachment) {
```

Comment:
This method name explains what it does, but doesnt really do a good job at giving specific context.

I vote for `getNameOf`.